src/subprocess.rs

```diff
@@ -0,0 +1,76 @@
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use Subprocess;
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use Cancellable;
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use Error;
+use ffi;
+use glib;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::ptr;
+
+pub trait SubprocessExtManual {
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn communicate_utf8_async<'a, P: Into<Option<String>>, Q: Into<Option<&'a Cancellable>>, R: FnOnce(Result<(String, String), Error>) + Send + 'static>(&self, stdin_buf: P, cancellable: Q, callback: R);
+
+    #[cfg(feature = "futures")]
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn communicate_utf8_async_future<P: Into<Option<String>>>(&self, stdin_buf: P) -> Box_<futures_core::Future<Item = (Self, (String, String)), Error = (Self, Error)>>;
+}
+
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+impl<O: IsA<Subprocess> + IsA<glib::object::Object> + Clone + 'static> SubprocessExtManual for O {
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn communicate_utf8_async<'a, P: Into<Option<String>>, Q: Into<Option<&'a Cancellable>>, R: FnOnce(Result<(String, String), Error>) + Send + 'static>(&self, stdin_buf: P, cancellable: Q, callback: R) {
+        let stdin_buf = stdin_buf.into();
+        let stdin_buf = stdin_buf.to_glib_none();
+        let cancellable = cancellable.into();
+        let cancellable = cancellable.to_glib_none();
+        let user_data: Box<Box<(R, *mut i8)>> = Box::new(Box::new((callback, stdin_buf.0)));
```

Comment:
The `stdin_buf` has to be the result of `to_glib_full()` as you want to keep it alive until the trampoline ran and then free it there.